### PR TITLE
Fix security vulnerabilities: sanitize ytdlp arguments and prevent XSS in useLinkify

### PR DESCRIPTION
**Pull request by independent-arg**
### Vulnerability

`ytdlp_runner.rs` was logging the full yt-dlp command line via `tracing::debug!` before execution in both `output()` and `spawn()`:

```rust
tracing::info!("Running command: yt-dlp {}", self.args.join(" "));
```

This exposed credentials in plain text, including:

--username / -u
--password / -p
--video-password
--ap-username / --ap-password (if used)
--proxy (may contain embedded credentials like http://user:pass@host)
--add-header (any header, but especially Authorization:Bearer and Cookie)
--cookies (file path, which may reveal user directory structure)
--extractor-args (can include API keys, e.g. youtube:api_key=...)
--sponsorblock-api (can include a personal API key)

### Impact

Secrets were written to local log files and could be transmitted to Sentry,
an external error-tracking service integrated with the application.
When the app reports an error, Sentry captures recent logs as context and sends
them to its servers, meaning credentials could leave the user's machine
without their knowledge.

**_This change only affects what is written to the logs. The command sent to the yt-dlp subprocess remains exactly the same as before, no functionality is altered_**

### Fix

Added a `sanitize_args_for_log` function that masks sensitive values with `[REDACTED]`
before logging. The actual arguments passed to yt-dlp are not modified.

```rust
// Before (vulnerable)
tracing::debug!("Running command: yt-dlp {}", self.args.join(" "));

// After (safe)
tracing::debug!(
  "Running command: yt-dlp {}",
  sanitize_args_for_log(&self.args)
);
```

### Testing

- Added `sanitize_args_redacts_sensitive_values` unit test that verifies
-- All sensitive flags are correctly masked.
-- The original values never appear in the sanitised output.
-- Non‑sensitive flags (like --url, --no-playlist, etc.) remain untouched.

### CODE REVIEW

**Comment by independent-arg** [comment]
I look forward to your review of this, @jely2002 . It's a very easy fix, perhaps it can be included in the next release.

**Comment by independent-arg** [comment]
I fixed an XSS vulnerability, too, in useLinkify.ts and added some tests to make everything more robust (I just happened to run into it unexpectedly).

File: src-tauri/src/runners/ytdlp_runner.rs [hunk 1]
```diff
@@ -384,6 +390,45 @@ impl<'a> YtdlpRunner<'a> {
   }
 }
 
+fn sanitize_args_for_log(args: &[String]) -> String {
```
Comment (jely2002) [inline src-tauri/src/runners/ytdlp_runner.rs]: I think this might be a little brittle. It might miss things when new flags get added and we forget this list exists hardcoded here.

Would it not be better to instead not log or send the command at all? Then we do not need to scrub anything from it in the first place.

In `src-tauri/src/lib.rs:251` we can use the `tracing_levels()` function, so the yt-dlp args don't get logged anymore. The log level for logging in production is already set to INFO using that function, so no commands get logged with normal release versions.

File: src-tauri/src/runners/ytdlp_runner.rs [hunk 2]
```diff
@@ -384,6 +390,45 @@ impl<'a> YtdlpRunner<'a> {
...
+      } else {
+        result.push(arg.clone());
+      }
+      continue;
+    }
+
+    if sensitive_flags.contains(&arg.as_str()) {
+      result.push(arg.clone());
+      if let Some(next) = iter.peek() {
+        if !next.starts_with('-') {
```
Comment (jely2002) [inline src-tauri/src/runners/ytdlp_runner.rs]: Technically if a secret would start with a '-' it would not get redacted here I think. But nonetheless check my other comment.

**Review by jely2002 — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution.

The linkify changes make total sense and look good. 
For the ytdlp sanitization, see my comment.

If we don't send the full command to Sentry and don't log it, we could choose to still log and sent a safe version like:

```
  tracing::info!(
    arg_count = self.args.len(),
    has_proxy = self.args.iter().any(|arg| arg == "--proxy" || arg.starts_with("--proxy=")),
    has_cookies = self.args.iter().any(|arg| arg == "--cookies" || arg.starts_with("--cookies=")),
    has_browser_cookies = self.args.iter().any(|arg| arg == "--cookies-from-browser" || arg.starts_with("--cookies-from-browser=")),
    has_auth = self.args.iter().any(|arg| {
      matches!(
        arg.as_str(),
        "--username" | "--password" | "--video-password" | "--add-header"
      ) || arg.starts_with("--username=")
        || arg.starts_with("--password=")
        || arg.starts_with("--video-password=")
        || arg.starts_with("--add-header=")
    }),
    "Running yt-dlp command"
  );
```

If it was only the linkify changes this would get merged, but now its blocked on the sanitization, which is okay I guess.

**Comment by independent-arg** [comment]
Thanks for the detailed feedback! both points made sense, so I went with your suggested approach instead of trying to patch the existing redaction logic.

I removed `sanitize_args_for_log` entirely and replaced it with a presence-only summary:

```rust
struct RunLogSummary {
  arg_count: usize,
  has_proxy: bool,
  has_cookies: bool,
  has_browser_cookies: bool,
  has_auth: bool,
}
```

`output()` and `spawn()` now call `log_run_summary(&self.args)`, which builds this summary and logs it via `tracing::info!`, no argument values are inspected or logged anymore, only whether certain flag categories are present.

This addresses both comments at once: since nothing about the actual value is ever read, there's no blacklist left to keep in sync with future flags, and the edge case where a secret starting with `-` slipped past redaction is no longer possible, there's no redaction step to slip past.

I also swapped the old `sanitize_args_redacts_sensitive_values` test for a few tests on `summarize_args_for_log`, including one that explicitly uses a dash-prefixed password value as a regression check for that edge case.

One thing I want to flag: the summary currently covers `has_proxy`, `has_cookies`, `has_browser_cookies`, and `has_auth`, but drops the visibility the old logging had for `--extractor-args` and `--sponsorblock-api` (both of which can carry embedded API keys). It's not a safety issue either way, just less observability than before. Happy to add `has_extractor_args` / `has_sponsorblock_api` as additional booleans if that's useful, let me know.

File: src-tauri/src/runners/ytdlp_runner.rs [hunk 3]
```diff
@@ -298,7 +298,7 @@ impl<'a> YtdlpRunner<'a> {
   }
 
   pub fn spawn(self) -> Result<(UnboundedReceiver<YtdlpCommandEvent>, YtdlpChild), String> {
-    tracing::debug!("Running command: yt-dlp {}", self.args.join(" "));
```
Comment (jely2002) [inline src-tauri/src/runners/ytdlp_runner.rs:301]: This line should stay for tracing::debug, so that in development we can still see full commands. Then in lib.rs we can set the sentry log level from TRACE to INFO so sentry never sends any debug messages.

https://github.com/jely2002/youtube-dl-gui/blob/4c9288c62968358d93442f1786f45a5e0469a5f2/src-tauri/src/lib.rs#L251 should use `tracing_levels()` instead of the hardcoded TRACE.

As its very useful to see the full logs in development!

Comment (independent-arg) [reply src-tauri/src/runners/ytdlp_runner.rs:301]: @jely2002 I've made that change. Thanks for reminding me about that debug line, it had slipped my mind.

**Review by jely2002 — APPROVED** [review]

**Comment by independent-arg** [comment]
@jely2002 Thanks for helping me with this as well. By the way, I’d like to suggest implementing a `SECURITY.md` file so that vulnerabilities (if any) can be reported privately (it would make things more secure) :)

**Comment by jely2002** [comment]
Probably a good point to add a `SECURITY.md` consider it done.
